feat: implement GEO extension with decToGMS and geoDistance functions

SashankBhamidi opened the pull request:
## Summary

Implements GEO extension for OpenRefine adding geographic coordinate processing functions.

### Functions

- `decToGMS(decimal [, 'lat'|'lng'])`: Converts decimal degrees to degrees/minutes/seconds format
- `geoDistance(lat1, lng1, lat2, lng2 [, unit])`: Calculates great circle distance using Haversine formula

### Implementation

- 24 unit tests, all passing
- Zero external dependencies
- Input validation with coordinate bounds checking
- Locale-safe decimal formatting
- Professional error handling

### Test Results

- Maven build: SUCCESS
- Test suite: 24/24 passing
- Linting: All checks pass
- No compilation errors

Resolves #6570


<!-- Issuehunt content -->

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#6570: New GEO Extension with commonly needed functions for GIS, geography, etc.](https://oss.issuehunt.io/repos/6220644/issues/6570)
---
</details>
<!-- /Issuehunt content-->

Abbe98 commented:
This is awesome! That said it should probably be in its own repository.

SoryRawyer commented:
Hi @SashankBhamidi, thanks for contributing these functions! I agree with @Abbe98 that this would work best in its own repository. Unfortunately, I don't know much about Issue Hunt so I'm probably not the best one to answer questions related to that. Regarding the extensions currently in the repository, my understanding is they're included in this repo due to some historical reasons that aren't entirely relevant any more. There's been some discussion about moving the wikibase extension to its own repository, for instance.

Regarding integrating this work into OpenRefine, if this is split into its own repository then I think there's no further work that needs to happen with the core OpenRefine codebase. Your extension registers its GREL functions and users install the extension themselves. The documentation for these functions would live with the extension code. You would be very welcome to open a PR to add your extension to [the list on the website](https://github.com/OpenRefine/openrefine.org/blob/master/src/pages/extensions.md).

SashankBhamidi commented:
Thanks @SoryRawyer!

I've followed your guidance and created the separate repository at https://github.com/SashankBhamidi/openrefine-geo-extension. 

The extension is completely self-contained and registers the GREL functions independently without needing any core codebase changes.

I also took your suggestion and opened a PR to add it to the extensions list: https://github.com/OpenRefine/openrefine.org/pull/486

One quick note: I added an empty commit here since this PR is linked to issue #6570, and the IssueHunt bounty system requires both the PR and issue to be merged and closed respectively for payout.

Thanks for pointing me in the right direction on OpenRefine's extension architecture!

SashankBhamidi commented:
I've updated the documentation to reflect that it's now an official OpenRefine extension housed under the OpenRefine organization.

Also updated the website PR (https://github.com/OpenRefine/openrefine.org/pull/486) and main PR to reference the OpenRefine org URL (https://github.com/OpenRefine/openrefine-geo-extension) instead of my personal repo.

SoryRawyer commented:
@SashankBhamidi apparently I don't have permission to fork the repository into the OpenRefine org. Presumably an admin needs to do that. I'll see if the permissions are any different for transferring the fork and, if I'm able, I'll give you temporary access to transfer the fork. I really appreciate your patience through all of this, hopefully we're close to wrapping this up!

SashankBhamidi commented:
> @SashankBhamidi apparently I don't have permission to fork the repository into the OpenRefine org. Presumably an admin needs to do that. I'll see if the permissions are any different for transferring the fork and, if I'm able, I'll give you temporary access to transfer the fork. I really appreciate your patience through all of this, hopefully we're close to wrapping this up!

Anything’s fine, I’ll be around for the next 120 minutes if you want to give me temporary/restricted access just for creating repos.

SashankBhamidi commented:
@SoryRawyer Otherwise, try forking it into your personal account, then detach the fork and transfer it from your profile to the org. That should do the trick I guess.

magdmartin commented:
the forked repo is available here https://github.com/OpenRefine/openrefine-geo-extension

SoryRawyer commented:
Thanks! I'll go ahead and:
1. merge this PR
2. contact GitHub support to see if I can detach the fork

SashankBhamidi commented:
I was just rewarded for a different bounty earlier, and in that case they rewarded me on IssueHunt **before** closing the issue.

Right now, since this issue was already closed, it isn’t letting my PR be reviewed/rewarded on the IssueHunt dashboard (as shown in the screenshot shared by @thadguidry).

The issue author doesn’t need to be the one to reward, it can be done by anyone with write access (maintainers/collaborators). 

The steps are:
1. Re-open the issue #6570
3. Reward my PR from the IssueHunt dashboard for this issue: [https://oss.issuehunt.io/r/OpenRefine/OpenRefine/issues/6570](https://oss.issuehunt.io/r/OpenRefine/OpenRefine/issues/6570).

This is the same procedure that was followed for a previous bounty in OpenRefine: [https://github.com/OpenRefine/OpenRefine/issues/3224#issuecomment-1627175544](https://github.com/OpenRefine/OpenRefine/issues/3224#issuecomment-1627175544).

Someone with access will need to follow this procedure to release the reward to me. Thanks!

SoryRawyer commented:
The IssueHunt page says there's an active bounty request, could that be interfering with the process?

thadguidry commented:
Likely it will need to be @tfmorris or @magdmartin who needs to login with their GitHub account into IssueHunt.io and reward you as accepting that PR 6432 that it seems to be showing in my view above as well.  I just dont have those 2 buttons likely because I am no longer part of the Core dev team and withdrew my rights.

Work with Martin or Tom to get rewarded.

magdmartin commented:
I was not able to find how to release the payout from issuehunt. I sent an email to their support.
